### PR TITLE
Make /etc/issue settable

### LINKED ISSUE #273: Make banner setable via pillar
As more orgs start using this project's contents, have been receiving out-of-band requests to be able to set org-specific banners via this project's automation-hooks

Will leave the current content as the banner-default but allow setting of local/override content via pillar

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #273

### CODE REVIEW

File: ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls [hunk 1]
```diff
@@ -16,7 +16,8 @@
 #################################################################
 {%- set stig_id = 'RHEL-07-010040' %}
 {%- set helperLoc = 'ash-linux/el7/STIGbyID/cat2/files' %}
-{%- set bannerText = 'ash-linux/el7/banner-consent_full.txt' %}
+{%- import_text "ash-linux/el7/banner-consent_full.txt" as default_banner %}
+{%- set bannerText = salt.pillar.get('ash-linux:lookup:banner-plain', default_banner) %}
```
Comment (lorengordon) [inline ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: What does "plain" refer to in this context? I'm not sure that's going to make any sense in a user's pillar. Just `banner` or `login-banner` or maybe `/etc/issue` (presuming `/` is ok) would seem to make more sense.

File: ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls [hunk 2]
```diff
@@ -26,8 +27,8 @@ script_{{ stig_id }}-describe:
 file_{{ stig_id }}:
   file.managed:
     - name: '/etc/issue'
-    - source: salt://{{ bannerText }}
     - user: root
     - group: root
     - mode: 0644
+    - contents: {{ bannerText|yaml(False)|indent(8) }}
```
Comment (lorengordon) [inline ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: Can you confirm this works both when using the bundled file `banner-consent_full.txt` as well as when set via pillar?

**Comment by ferricoxide** [comment]
> There's a bunch of extra files in this PR. Not sure where they're coming from, but definitely unrelated.

As noted in their commit message, they were a "while we're writing to this project, add them" content. They're templates for future PRs and Issues.

File: ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls (hunk 2 above)
Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: When I nuke the `banner-plain` key from the pillar and re-run:

~~~
----------
          ID: file_RHEL-07-010040
    Function: file.managed
        Name: /etc/issue
      Result: True
     Comment: File /etc/issue updated
     Started: 20:27:11.254528
    Duration: 20.974 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,9 +1,13 @@
                  -This is a super secret-squirrel system!
                  +You are accessing a U.S. Government (USG) Information System (IS) that is provided for USG-authorized use only.

                  -Shoo: g'away!
                  +By using this IS (which includes any device attached to this IS), you consent to the following conditions:

                  -You're being watched!
                  +-The USG routinely intercepts and monitors communications on this IS for purposes including, but not limited to, penetration testing, COMSEC monitoring, network operations and defense, personnel misconduct (PM), law enforcement (LE), and counterintelligence (CI) investigations.

                  -Don't say we didn't warn you!
                  +-At any time, the USG may inspect and seize data stored on this IS.

                  -Gonna call the cops on you if you don't go away!
                  +-Communications using, or data stored on, this IS are not private, are subject to routine monitoring, interception, and search, and may be disclosed or used for any USG-authorized purpose.
                  +
                  +-This IS includes security measures (e.g., authentication and access controls) to protect USG interests--not for your personal benefit or privacy.
                  +
                  +-Notwithstanding the above, using this IS does not constitute consent to PM, LE or CI investigative searching or monitoring of the content of privileged communications, or work product, related to personal representation or services by attorneys, psychotherapists, or clergy, and their assistants. Such communications and work product are private and confidential. See User Agreement for details.

Summary for local
------------
Succeeded: 2 (changed=2)
Failed:    0
------------
Total states run:     2
Total run time:  55.775 ms
[root@ash-issue273 pillar]# cat /etc/issue
You are accessing a U.S. Government (USG) Information System (IS) that is provided for USG-authorized use only.

By using this IS (which includes any device attached to this IS), you consent to the following conditions:

-The USG routinely intercepts and monitors communications on this IS for purposes including, but not limited to, penetration testing, COMSEC monitoring, network operations and defense, personnel misconduct (PM), law enforcement (LE), and counterintelligence (CI) investigations.

-At any time, the USG may inspect and seize data stored on this IS.

-Communications using, or data stored on, this IS are not private, are subject to routine monitoring, interception, and search, and may be disclosed or used for any USG-authorized purpose.

-This IS includes security measures (e.g., authentication and access controls) to protect USG interests--not for your personal benefit or privacy.

-Notwithstanding the above, using this IS does not constitute consent to PM, LE or CI investigative searching or monitoring of the content of privileged communications, or work product, related to personal representation or services by attorneys, psychotherapists, or clergy, and their assistants. Such communications and work product are private and confidential. See User Agreement for details.
~~~

So, answer _appears_ to be "yes, works both ways." :smile:

File: ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls (hunk 1 above)
Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: `plain` means "as used outside of the `gconf` context. The other context is `banner-gconf` (not included in this PR since this PR's issue was only for `/etc/issue` and, to the best of my knowledge, we haven't tested _any_ EL7 hardening for GDM since those states were written (and were only minimally tested - i.e., "does `gconf` puke if I use a particular method for passing content in" vice "does the window GDM pops up actually look right").

GDM wants things in a different format from what `/etc/issue` wants them in, thus the two banner `.txt` files that have been in the project (since the states were first written)

Basically, there's a couple types of banners – most services (now, with the exception of GDM) simply use whatever's in `/etc/issue`.

**Comment by ferricoxide** [comment]
@lorengordon 

Updated the pillar file in 2604c3c86d20f8c05c3440b6db1fbf1a5d5a9dbe. Did not update documentation as no updatable documentation exists in _this_ project.

File: ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls (hunk 1 above)
Comment (lorengordon) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: Ok, so `login-banner` still makes more sense as a key-name. And for `gconf`, if there is ever a feature request for it, could be similarly clear `login-banner-gconf`...

Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: I just _haaaaaate_ having two key-names where a second key-name is an extension of a prior key-name: makes future (potential) string-matching more of a pain in the ass (since `login-banner` is a substring-match to _both_ `login-banner`  and `login-banner-gconf`).

Comment (lorengordon) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: It's a dictionary data structure. Not sure what the use case would be to match substrings on a key name. Just load it and get the key name. Could also create subkeys:

```
login-banners:
  '/etc/issue': <str>
  gconf: <str>
```

Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: Yeah. Think I'd prefer the sub-key approach. Makes more sense, to me, mnemonically.

Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/RHEL-07-010040.sls]: Sub-key approach implemented via 8e353511da83b682d7ce6ea2e8e5e45ac08bbe83

**Comment by ferricoxide** [comment]
@lorengordon alright, _should_ be good to go. Sub-keys implemented.
